DCode.Services/Common/CommonService.cs

```diff
@@ -581,5 +583,16 @@ public List<string> GetFINotificationRecipientsForServiceLine(int serviceLineId)
                 ?.Split(',')
                 ?.ToList();
         }
+
+        public void SetOfferingCodeAndInsertIntoDb(IEnumerable<offering> offerings)
+        {

```

Comment:
Can you please optimise this code. Don't use ToList() with each reference to offerings. Please set it in local variable once before going into loop and use that temp variable.